### PR TITLE
[examiner] Adds permission check when adding examiner to site.

### LINKED ISSUE #9149: [examiner] I was able to create an examiner for a site I don't belong to
I accessed the examiner module with a user that only has the `Examiner: Add and Certify Examiners - Own Sites` permission and belongs to site `Rome`. I was able to create an examiner for site `DCC` but I shouldn't have been able to (I can't even see the examiners from `DCC`...)

### PR DESCRIPTION
**Pull request by racostas**
## Brief summary of changes
Permissions are check now before allowing to add examiner to site. 
Before this PR, a user with only the permission `Examiner: Add and Certify Examiners - Own Sites` was able to add examiners to sites it don't belongs to. Now this feature is only granted to users with the corresponding level of permission.

#### Testing instructions (if applicable)

1. With a user that only have the permission `Examiner: Add and Certify Examiners - Own Sites`  try to create a new examiner in `MainMenu->Clinical->Examinier`
2.  When clicking the Add Examiner button the modal windows that opened should only list the sites the current user belongs to.
3.  See image as reference.
image
4. Only for the listed sites the current user should be able to add an examiner.
#### Link(s) to related issue(s)

* Resolves #9149

### CODE REVIEW

**Comment by sruthymathew123** [comment]
Tested and looks good.

**Comment by sruthymathew123** [comment]
@driusan Tested and looks good on my end.

File: modules/examiner/php/addexaminer.class.inc [hunk 1]
```diff
@@ -64,6 +64,21 @@ class AddExaminer extends \NDB_Page
         $fullName = $values['addName'] ?? '';
         $siteID   = $values['addSite'] ?? '';
 
+        // check for site permissions
+        $user = \User::singleton();
+        if (!$user->hasAnyPermission(
+            [
+                'superuser',
```
Comment (driusan) [inline modules/examiner/php/addexaminer.class.inc:71]: superuser should not be here. (hasAnyPermission is a wrapper around hasPermission which already handles it.)

File: modules/examiner/php/addexaminer.class.inc [hunk 2]
```diff
@@ -64,6 +64,21 @@ class AddExaminer extends \NDB_Page
...
+                'examiner_multisite',
+            ]
+        )
+            && !in_array($siteID, $user->getCenterIDs())
+        ) {
+            return  new \LORIS\Http\Response\JSON\Conflict(
```
Comment (driusan) [inline modules/examiner/php/addexaminer.class.inc:77]: This situation is a `403 Forbidden` not a `409 Conflict` situation. There is no conflict with editing the state of the resource, access is forbidden to it.

**Pull request merged**